src/huggingface_hub/file_download.py

```diff
@@ -329,6 +326,15 @@ def _get_file_length_from_http_response(response: requests.Response) -> Optional
         `int` or `None`: The length of the file in bytes, or None if not available.
     """
 
+    # If HTTP response contains compressed body (e.g. gzip), the `Content-Length` header will
+    # contain the length of the compressed body, not the uncompressed file size.
+    # And at the start of transmission there's no way to know the uncompressed file size for gzip,
+    # thus we return None in that case.
+    content_encoding = response.headers.get("Content-Encoding")
+    if (content_encoding is not None) and ("identity" not in content_encoding.lower()):
+        # gzip/br/deflate/zstd etc
+        return None
```

Comment:
slight preference for better readability
```suggestion
    content_encoding = response.headers.get("Content-Encoding", "identity").lower()
    if content_encoding != "identity":
        # gzip/br/deflate/zstd etc
        return None
```